alf/algorithms/entropy_target_algorithm.py

```diff
@@ -214,61 +203,62 @@ def adjust_alpha(self, entropy):
         avg_entropy = self._avg_entropy.average(entropy)
 
         def _init_entropy():
-            self._max_entropy.assign(
-                tf.minimum(0.8 * avg_entropy, avg_entropy / 0.8))
-            self._stage.assign_add(1)
+            self._max_entropy.fill_(min(0.8 * avg_entropy, avg_entropy / 0.8))
+            self._stage.add_(1)
 
         def _init():
             below = avg_entropy < self._max_entropy
-            increasing = tf.cast(avg_entropy > prev_avg_entropy, tf.float32)
+            increasing = (avg_entropy > prev_avg_entropy).type(torch.float32)
             # -1 * increasing + 0.5 * (1 - increasing)
             update_rate = (
                 0.5 - 1.5 * increasing) * self._very_slow_update_rate
-            self._stage.assign_add(tf.cast(below, tf.int32))
-            self._log_alpha.assign(
-                tf.maximum(self._log_alpha + update_rate,
-                           np.float32(self._min_log_alpha)))
+            self._stage.add_(below.type(torch.int32))
+            self._log_alpha.fill_(
+                max(self._log_alpha + update_rate,
+                    np.float32(self._min_log_alpha)))
 
         def _free():
             crossing = avg_entropy < self._target_entropy
-            self._stage.assign_add(tf.cast(crossing, tf.int32))
+            self._stage.add_(crossing.type(torch.int32))
 
         def _adjust():
-            previous_above = tf.cast(self._stage, tf.bool)
+            previous_above = self._stage.type(torch.bool)
             above = avg_entropy > self._target_entropy
-            self._stage.assign(tf.cast(above, tf.int32))
+            self._stage.fill_(above.type(torch.int32))
             crossing = above != previous_above
             update_rate = self._update_rate
-            update_rate = tf.where(crossing, 0.9 * update_rate, update_rate)
-            update_rate = tf.maximum(update_rate, self._slow_update_rate)
-            update_rate = tf.where(entropy < self._fast_stage_thresh,
-                                   np.float32(self._fast_update_rate),
-                                   update_rate)
-            self._update_rate.assign(update_rate)
-            above = tf.cast(above, tf.float32)
+            update_rate = torch.where(crossing, 0.9 * update_rate, update_rate)
```

Comment:
the arguments for where should be Tensor

Reply:
Done.